src/lib.rs

```diff
@@ -145,8 +144,10 @@ pub trait MiniscriptKey: Clone + Eq + Ord + fmt::Debug + fmt::Display + hash::Ha
 
     /// Computes the size of a public key when serialized in a script,
     /// including the length bytes
-    fn serialized_len(&self) -> usize {
-        if self.is_uncompressed() {
+    fn serialized_len<Ctx: ScriptContext>(&self) -> usize {
```

Comment:
What if we inverted this function, put it on the `Ctx` object like
```
fn pk_len<Pk: MiniscriptKey>(_pk: &Pk) -> usize
```
and have the Segwit context return 34 always, Tap context return 33 always, and the bare context would look at `pk.is_compressed()`.

?